docs/java/how-to-develop-an-activity-definition-in-java.md

```diff
@@ -0,0 +1,110 @@
+title: How to develop an Activity Definition in Java
+sidebar_label: Activity Definition
+description: In the Temporal Java SDK programming model, Activities are classes which implement the Activity Interface.
+tags:
+  - java
+  - developer-guide
+---
+
+An [Activity Definition](/docs/concepts/what-is-an-activity) is a combination of the Temporal Java SDK [Activity](https://www.javadoc.io/static/io.temporal/temporal-sdk/0.19.0/io/temporal/activity/Activity.html) Class implementing a specially annotated interface.
+
+## Activity interface
+
+Activity interface is a Java interface and is annotated with the `@ActivityInterface` annotation.
+
+```java
+@ActivityInterface
+public interface GreetingActivities {
+    String composeGreeting(String greeting, String language);
+}
+```
+
+Each method defined in the Actvity interface defines a separate Activity method.
+You can annotate each method in the Activity interface with the `@ActivityMethod` annotation, but this is completely optional. The following example uses the `@ActivityMethod` annotation for the method defined in the previous example.
+
+```java
+@ActivityInterface
+public interface GreetingActivities {
+    @ActivityMethod()
+    String composeGreeting(String greeting, String language);
+}
+```
+
+The Activity Type defaults to method name, with the first letter of the method name capitalized. In the previous example, the Activity Type defaults to `ComposeGreeting`.
+
+To overwrite this default naming and assign a custom Activity Type, use the `@ActivityMethod` annotation with the `name` parameter. In the following example, the Activity Type is set to "greet".
+
+```java
+@ActivityInterface
+public interface GreetingActivities {
+    @ActivityMethod(name = "greet")
+    String composeGreeting(String greeting, String language);
+}
+```
+
+You can also define a prefix for all of your Activity Types using the `namePrefix` parameter with the `@ActivityInterface` annotation. The following example shows a `namePrefix` parameter applied to the `@ActivityInterface`, and two Activity methods, of which one is defined using the `@ActivityMethod` annotation.
+
+```java
+@ActivityInterface(namePrefix = "A_")
+Public interface MyActivity {
+    String doSomething(String input);
+
+  @ActivityMethod(name = "abc")
+  String doSomethingElse();
+}
+```
+
+In this example, the Activity type for the first method is set to "A_DoSomething". The Activity type for the method annotated with `@ActivityMethod` is set to "A_abc".
+
+## Activity implementation
+
+An Activity implementation is Java class which implements an Activity Interface.
+Example:
+
+```java
+  public class MultiLanguageGreeting implements GreetingsActivities {
+    private final TranslationService translationService;
+
+    public MultiLanguageGreeting(TranslationService translationService) {
+      this.translationService = translationService;
+    }
+
+    @Override
+    public String composeGreeting(String greeting, String language) {
```

Comment:
I think we are missing information about method parameters and return values.

I think similarly to Go, in Java we recommend passing a custom param type so that you can iterate on the individual parameters without breaking the method signature.